### PR TITLE
EnableAzurePipelinesReporter for helix

### LINKED ISSUE #7368: [Helix] Reenable AzDo error reporting for helix tests
Turning on EnableAzurePipelinesReporter results in an error currently

```
2019-02-06T08:43:00.3652531Z D:\a\1\s\.nuget\packages\microsoft.dotnet.helix.sdk\2.0.0-beta.19073.6\tools\azure-pipelines\AzurePipelines.props(27,5): 
error : Azure Pipelines SYSTEM_ACCESSTOKEN variable not found. Either the build is not running in Azure Pipelines, or the current step is not configured correctly. 
Please configure this step with the System.AccessToken variable, see: https://docs.microsoft.com/en-us/azure/devops/pipelines/process/variables?view=vsts&tabs=yaml,batch#systemaccesstoken [D:\a\1\s\test\helix.proj]
```

### LINKED ISSUE #8087: Enable helix VSTS test reporting
https://github.com/dotnet/arcade/blob/master/src/Microsoft.DotNet.Helix/Sdk/Readme.md

```
    <!-- Enable reporting of test results to azure dev ops -->
    <EnableAzurePipelinesReporter>false</EnableAzurePipelinesReporter>
```

cc @anurse

### PR DESCRIPTION
**Pull request by pakrym**
Fix: https://github.com/aspnet/AspNetCore/issues/7368
Fix: https://github.com/aspnet/AspNetCore/issues/8087

### CODE REVIEW

**Comment by anurse** [comment]
Test Run Aborted... \*shakes fist*

**Comment by anurse** [comment]
I don't see any results :(

**Comment by pakrym** [comment]
Yep, me neither. Investigating.

**Comment by HaoK** [comment]
adding @MattGal in case there's anything else we might need to enable other than just the flag

**Comment by pakrym** [comment]
I wonder if some of our custom logic might be the cause.

**Comment by HaoK** [comment]
The good news is that it doesn't blow up with the issue I saw when I tried it the first time: https://github.com/aspnet/AspNetCore/issues/7368

**Comment by HaoK** [comment]
We also don't have xunit reporter turned on, not sure if that is a requirement for the azdo helix stuff

https://github.com/dotnet/arcade/blob/2ec202777efff9fd36518804438f7272a34040f4/eng/common/templates/steps/send-to-helix.yml#L20

**Comment by pakrym** [comment]
@alexperovich any idea why only random small subset of test results got uploaded to VSTS?

File: eng/helix/vstest/runtests.cmd [hunk 1]
```diff
@@ -23,6 +23,12 @@ if %errorlevel% equ 0 (
...
 )
 
-%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --logger:console;verbosity=normal
+%DOTNET_ROOT%\dotnet vstest %target% --logger:trx;LogFileName=test-results.xml --logger:console;verbosity=normal
 
+set _ERRORLEVEL=%ERRORLEVEL%
 
+rem Normalize test result file name
+
+ren TestResults\test-results*.xml test-results.xml
```
Comment (dougbu) [inline eng/helix/vstest/runtests.cmd]: Is this still needed with the `LogFileName=test-results.xml` change just above?

File: eng/helix/vstest/runtests.sh [hunk 2]
```diff
@@ -67,4 +67,12 @@ if grep -q "Exception thrown" discovered.txt; then
...
 fi
 
-$DOTNET_ROOT/dotnet vstest $1 --logger:trx
+$HELIX_CORRELATION_PAYLOAD/sdk/dotnet vstest $1 '--logger:trx;LogFileName=test-results.xml' '--logger:console;verbosity=normal'
+
+test_exit_code=$?
+
+# Normalize test result file name
+
+mv TestResults/test-results*.xml test-results.xml
```
Comment (dougbu) [inline eng/helix/vstest/runtests.sh]: Again, is this still needed?

**Comment by pakrym** [comment]
XUnit results are not published to MC either

File: eng/helix/vstest/runtests.cmd (hunk 1 above)
Comment (pakrym) [reply eng/helix/vstest/runtests.cmd]: Yes, vstest inserts timestamp into `test-results.xml`

**Comment by HaoK** [comment]
Nice so it looks like azdo flags missing test results as an error, so maybe we don't need the explicit test discovery check I just added to our scripts anymore if it fails the run with 
```
E:\A\_work\3\s\.nuget\packages\microsoft.dotnet.helix.sdk\2.0.0-beta.19105.2\tools\azure-pipelines\AzurePipelines.MultiQueue.targets(39,5): error : Test run 2576054 has one or more failing tests. [E:\A\_work\3\s\eng\helix\helix.proj]
```

**Comment by pakrym** [comment]
> Nice so it looks like azdo flags missing test results as an error, so maybe we don't need the explicit test discovery check I just added to our scripts anymore if it fails the run with

It doesn't have reported tests because of your check that prevented the reporter from running.

**Comment by pakrym** [comment]
Tests are not reported because RTX and xUnit XML are not the same.

**Comment by pakrym** [comment]
Blocked https://github.com/dotnet/arcade/issues/2161

**Comment by natemcmaster** [comment]
This should be unblocked now (https://github.com/dotnet/arcade/issues/2286), but @pakrym has switched teams. @JunTaoLuo  can you take a look at completing this?

**Comment by natemcmaster** [comment]
Looks good so far. The helix test is still running, but I already see the UI preparing to show results

image

**Comment by dougbu** [comment]
Even more than the usual number of warnings in the Helix build output. And, it looks like the Helix run caught some flaky tests. Either fix some of the warnings and update this branch or retry AspNetCore-helix-test before we merge.

**Comment by HaoK** [comment]
Hrm helix run looks pretty normal with only interop and one IIS test failing, but the test results all seem to have failed with missing result files, or am I looking at the wrong view?

**Comment by JunTaoLuo** [comment]
I think the warnings are due to https://github.com/dotnet/arcade/issues/1605. I'm not sure there's a fix for it. I'll retry the tests though.

**Comment by JunTaoLuo** [comment]
@HaoK It seems like the `'--logger:trx;LogFileName=test-results.xml'` argument is being ignored since the build complains: `No results file found in any of the following formats: xunit` Any way to figure out why the test results are not being logged?

File: eng/helix/vstest/runtests.cmd [hunk 3]
```diff
@@ -36,7 +36,7 @@ REM only puts the explicit filter traits the user provided in
 REM Filter syntax: https://github.com/Microsoft/vstest-docs/blob/master/docs/filter.md
 set NONFLAKY_FILTER="Flaky:All!=true&Flaky:Helix:All!=true&Flaky:Helix:Queue:All!=true&Flaky:Helix:Queue:%HELIX%!=true"
 echo Running non-flaky tests.
-%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --logger:console;verbosity=normal --TestCaseFilter:%NONFLAKY_FILTER%
+%DOTNET_ROOT%\dotnet vstest %target% --logger:trx;LogFileName=test-results.xml --TestCaseFilter:%NONFLAKY_FILTER%
```
Comment (natemcmaster) [inline eng/helix/vstest/runtests.cmd]:
```suggestion
%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --TestCaseFilter:%NONFLAKY_FILTER%
```

No need to change the file name manually. https://github.com/dotnet/arcade/pull/2446/ implemented support for .trx files, and they will scan for any file named *.trx which is produced in the working directory (or subdirs).

File: eng/helix/vstest/runtests.cmd [hunk 4]
```diff
@@ -45,11 +45,15 @@ if errorlevel 1 (
 
 set FLAKY_FILTER="Flaky:All=true|Flaky:Helix:All=true|Flaky:Helix:Queue:All=true|Flaky:Helix:Queue:%HELIX%=true"
 echo Running known-flaky tests.
-%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --logger:console;verbosity=normal --TestCaseFilter:%FLAKY_FILTER%
+%DOTNET_ROOT%\dotnet vstest %target% --logger:trx;LogFileName=test-results.xml --TestCaseFilter:%FLAKY_FILTER%
```
Comment (natemcmaster) [inline eng/helix/vstest/runtests.cmd]:
```suggestion
%DOTNET_ROOT%\dotnet vstest %target% --logger:trx --TestCaseFilter:%FLAKY_FILTER%
```

File: eng/helix/vstest/runtests.cmd [hunk 5]
```diff
@@ -45,11 +45,15 @@ if errorlevel 1 (
...
 if errorlevel 1 (
     echo Failure in flaky test 1>&2
     REM DO NOT EXIT and DO NOT SET EXIT_CODE to 1
 )
 
+rem Normalize test result file name
+
+ren TestResults\test-results*.xml test-results.xml
```
Comment (natemcmaster) [inline eng/helix/vstest/runtests.cmd]:
```suggestion
```

Can be removed

File: eng/helix/vstest/runtests.sh [hunk 6]
```diff
@@ -73,19 +73,22 @@ fi
 # Filter syntax: https://github.com/Microsoft/vstest-docs/blob/master/docs/filter.md
 NONFLAKY_FILTER="Flaky:All!=true&Flaky:Helix:All!=true&Flaky:Helix:Queue:All!=true&Flaky:Helix:Queue:$HELIX!=true"
 echo "Running non-flaky tests."
-$DOTNET_ROOT/dotnet vstest $1 --logger:trx --TestCaseFilter:"$NONFLAKY_FILTER"
+$DOTNET_ROOT/dotnet vstest $1  '--logger:trx;LogFileName=test-results.xml' --TestCaseFilter:"$NONFLAKY_FILTER"
```
Comment (natemcmaster) [inline eng/helix/vstest/runtests.sh]:
```suggestion
$DOTNET_ROOT/dotnet vstest $1  --logger:trx --TestCaseFilter:"$NONFLAKY_FILTER"
```

File: eng/helix/vstest/runtests.sh [hunk 7]
```diff
@@ -73,19 +73,22 @@ fi
...
 nonflaky_exitcode=$?
 if [ $nonflaky_exitcode != 0 ]; then
     echo "Non-flaky tests failed!" 1>&2
     # DO NOT EXIT
 fi
-
 FLAKY_FILTER="Flaky:All=true|Flaky:Helix:All=true|Flaky:Helix:Queue:All=true|Flaky:Helix:Queue:$HELIX=true"
 echo "Running known-flaky tests."
-$DOTNET_ROOT/dotnet vstest $1 --logger:trx --TestCaseFilter:"$FLAKY_FILTER"
+$DOTNET_ROOT/dotnet vstest $1  '--logger:trx;LogFileName=test-results.xml' --TestCaseFilter:"$FLAKY_FILTER"
```
Comment (natemcmaster) [inline eng/helix/vstest/runtests.sh]:
```suggestion
$DOTNET_ROOT/dotnet vstest $1  --logger:trx --TestCaseFilter:"$FLAKY_FILTER"
```

File: eng/helix/vstest/runtests.sh [hunk 8]
```diff
@@ -73,19 +73,22 @@ fi
...
 if [ $? != 0 ]; then
     echo "Flaky tests failed!" 1>&2
     # DO NOT EXIT
 fi
 
+# Normalize test result file name
+
+mv TestResults/test-results*.xml test-results.xml
```
Comment (natemcmaster) [inline eng/helix/vstest/runtests.sh]:
```suggestion
```

Can be removed

**Comment by alexperovich** [comment]
If you want to use the trx support in the helix sdk the file must be named `<someting>.trx`. If you output `test-results.xml` then it will try to parse it as xunit, even if it is trx.

That being said, you are using an ancient version of the helix sdk, it doesn't have support for trx at that version. You are using `2.0.0-beta.19105.2` the latest is `2.0.0-beta.19219.3`

**Review by natemcmaster — DISMISSED** [review]
@JunTaoLuo feel free to take on applying these changes. It looks like the problem with the parsing is that Helix is expecting the file to end in .trx, and expecting files named `test-results.xml` to be in the Xunit format (which we're not using)

**Comment by HaoK** [comment]
sounds like its a good time to update to latest helix sdk in this PR

**Comment by JunTaoLuo** [comment]
Hmm still didn't quite work correctly:
```
Searching '/home/helixbot/work/6ca299fc-f8a4-4299-8feb-55f2c18055fb/Work/4b3733d1-f767-44f4-ae46-ec11ea1eb7c8/Exec' for test results files
Searching '/home/helixbot/work/6ca299fc-f8a4-4299-8feb-55f2c18055fb/Work/4b3733d1-f767-44f4-ae46-ec11ea1eb7c8/Exec/..' for log files
Found log '/home/helixbot/work/6ca299fc-f8a4-4299-8feb-55f2c18055fb/Work/4b3733d1-f767-44f4-ae46-ec11ea1eb7c8/Exec/../console.3760326c.log'
Uri 'https://helixexternalresults.blob.core.windows.net/aspnet-aspnetcore-refs-pull-8094-merge-6ca299fcf8a442998f/Interop.FunctionalTests-netcoreapp3.0/console.3760326c.log?sv=2017-07-29&sr=c&sig=eLwi4dGkARsc3ryo8jKod0r07kZN5DaGu2B4u69aDTQ%3D&se=2019-05-01T07%3A34%3A52Z&sp=rl'
Generated log list: <ul><li><a href='https://helixexternalresults.blob.core.windows.net/aspnet-aspnetcore-refs-pull-8094-merge-6ca299fcf8a442998f/Interop.FunctionalTests-netcoreapp3.0/console.3760326c.log?sv=2017-07-29&sr=c&sig=eLwi4dGkARsc3ryo8jKod0r07kZN5DaGu2B4u69aDTQ%3D&se=2019-05-01T07%3A34%3A52Z&sp=rl' target='_blank'>console.3760326c.log</a></li></ul>
No results file found in any of the following formats: xunit, junit, trx
```

**Comment by HaoK** [comment]
Well I don't know about those errors in the log, but the tests tab look sorta reasonable now, it shows the correct failing tests dotnet watcher (for a flaky test failure), and the Interop.Functionals which fail during discovery so there is no trx log.

**Comment by JunTaoLuo** [comment]
Hmm so all of the InMemory.FunctionalTests have this in their logs:
```
2019-04-21 00:35:38,783: ERROR: xunit-reporter(69): main: Unable to report xunit results: no test results xml file found.
+ exit 0
```
But are finishing successfully, except for Debian.8.Amd64.Open which fails. But maybe that's because of
```
+ /usr/bin/python /home/helixbot/work/6ca299fc-f8a4-4299-8feb-55f2c18055fb/Payload/xunit-reporter.py
/usr/local/lib/python2.7/dist-packages/cryptography/hazmat/bindings/openssl/binding.py:163: CryptographyDeprecationWarning: OpenSSL version 1.0.1 is no longer supported by the OpenSSL project, please upgrade. A future version of cryptography will drop support for it.
  utils.CryptographyDeprecationWarning
```

As for the Interop.FunctionalTests:
```
[xUnit.net 00:00:00.60] Interop.FunctionalTests: Exception thrown during theory discovery on 'Interop.FunctionalTests.H2SpecTests.RunIndividualTestCase'; falling back to single test case.
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.ComponentModel.Win32Exception: Permission denied
   at System.Diagnostics.Process.ForkAndExecProcess(String filename, String[] argv, String[] envp, String cwd, Boolean redirectStdin, Boolean redirectStdout, Boolean redirectStderr, Boolean setCredentials, UInt32 userId, UInt32 groupId, UInt32[] groups, Int32& stdinFd, Int32& stdoutFd, Int32& stderrFd, Boolean usesTerminal, Boolean throwOnNoExec)
   at System.Diagnostics.Process.StartCore(ProcessStartInfo startInfo)
   at System.Diagnostics.Process.Start()
   at System.Diagnostics.Process.Start(ProcessStartInfo startInfo)
   at Interop.FunctionalTests.H2SpecCommands.EnumerateTestCases() in /_/src/Servers/Kestrel/test/Interop.FunctionalTests/H2SpecCommands.cs:line 42
   at Interop.FunctionalTests.H2SpecTests.get_H2SpecTestCases() in /_/src/Servers/Kestrel/test/Interop.FunctionalTests/H2SpecTests.cs:line 63
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, BindingFlags invokeAttr, Binder binder, Object[] index, CultureInfo culture)
   at System.Reflection.RuntimePropertyInfo.GetValue(Object obj, Object[] index)
   at Xunit.MemberDataAttributeBase.GetData(MethodInfo testMethod) in C:\Dev\xunit\xunit\src\xunit.core\MemberDataAttributeBase.cs:line 66
   at Xunit.Sdk.DataDiscoverer.GetData(IAttributeInfo dataAttribute, IMethodInfo testMethod) in C:\Dev\xunit\xunit\src\xunit.core\Sdk\DataDiscoverer.cs:line 25
   at Xunit.Sdk.TheoryDiscoverer.Discover(ITestFrameworkDiscoveryOptions discoveryOptions, ITestMethod testMethod, IAttributeInfo theoryAttribute) in C:\Dev\xunit\xunit\src\xunit.execution\Sdk\Frameworks\TheoryDiscoverer.cs:line 212
    Http2
    RunIndividualTestCase
+ export _commandExitCode=1
+ _commandExitCode=1
```
That still seems like a mystery.

File: eng/helix/vstest/runtests.sh [hunk 9]
```diff
@@ -73,16 +73,15 @@ fi
 # Filter syntax: https://github.com/Microsoft/vstest-docs/blob/master/docs/filter.md
 NONFLAKY_FILTER="Flaky:All!=true&Flaky:Helix:All!=true&Flaky:Helix:Queue:All!=true&Flaky:Helix:Queue:$HELIX!=true"
 echo "Running non-flaky tests."
-$DOTNET_ROOT/dotnet vstest $1 --logger:trx --TestCaseFilter:"$NONFLAKY_FILTER"
+$DOTNET_ROOT/dotnet vstest $1  '--logger:trx;LogFileName=test-results.xml' --TestCaseFilter:"$NONFLAKY_FILTER"
```
Comment (natemcmaster) [inline eng/helix/vstest/runtests.sh]:
```suggestion
$DOTNET_ROOT/dotnet vstest $1  --logger:trx --TestCaseFilter:"$NONFLAKY_FILTER"
```

One more

**Comment by natemcmaster** [comment]
I would have expected more test results than this. It looks like a few random tests were reported, but we have way more than 387 tests

image
image

**Comment by alexperovich** [comment]
@JunTaoLuo that xunit-reporter error is not related to azure pipelines test reporting. That is the legacy thing that reports xunit results to mission control. You can turn it off with the EnableXUnitReporter property.

**Comment by JunTaoLuo** [comment]
After investigating with @alexperovich, it seems like we need to run an additional chmod +x to run the H2Spec executable. While I'm not sure why that is, it seems like the tests are mostly running now and the results are being reported. There's one failing test which I will rerun since it seems like it was just flaky.

**Comment by natemcmaster** [comment]
We have about 1/3 as many tests reported on Helix. Are we really skipping that much on Helix?

### Helix
image

### Normal
image

**Comment by HaoK** [comment]
Hrm that delta does seem really large, let me see if I can find the biggest diffs

**Comment by natemcmaster** [comment]
I wonder if xunit theories are only reported as a single test case, instead of one per theory member….

**Comment by anurse** [comment]
> I wonder if xunit theories are only reported as a single test case, instead of one per theory member….

As long as the theory data is serializable, they are reported as separate. If it's not serializable, it'll group them up in to a single test case.

**Comment by Tratcher** [comment]
Ah, many of Kestrel's tests do not use serializable data.

**Comment by HaoK** [comment]
We must be missing a large group of tests, since I looked at one test, Authorization for example, and there's only (80 mac+80 linux+49 windows) on aspnet-ci, and on helix-test, I see 80 per queue (although windows is missing which is odd)

**Comment by HaoK** [comment]
Yeah so first thing that stands out, only 157 tests were reported on windows :/

**Comment by HaoK** [comment]
Missing MVC tests account for like 30k tests alone (12k mac/12k linux/8k windows on aspnetci) vs ~35-171 on helix queues.

They ran, for example https://mc.dot.net/#/user/aspnetcore/pr~2Faspnet~2Faspnetcore/ci/20190430.1/workItem/Microsoft.AspNetCore.Mvc.Core.Test-netcoreapp3.0/wilogs

shows 4.2k MVC core tests running alone on windows, but these aren't getting reported/picked up properly

**Comment by JunTaoLuo** [comment]
I'll see if I can get the trx file from the helix runs to upload to devops so we can compare.

**Comment by HaoK** [comment]
Could it be the additional flaky helix test run that's messing things up when it produces a second trx file?  Maybe we can we try removing that additional pass from the runtest.cmd/sh and see if that affects the count?

**Comment by JunTaoLuo** [comment]
Hmm this seems suspicious in the log:
```
Found results file /Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Work/3794520e-9043-4e3c-bf3d-120e76f5d6ef/Exec/TestResults/_dci-mac-build-080_2019-04-30_01_55_20_332.trx with format trx
Uploading results in batches of size 1000
Traceback (most recent call last):
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/run.py", line 95, in <module>
    main()
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/run.py", line 84, in main
    for b in batches:
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/helpers.py", line 13, in batch
    for item in iterable:
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/test_results_reader/__init__.py", line 94, in <genexpr>
    return (add_logs(tr, log_list) for tr in f.read_results(file_path))
  File "/Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/reporter/formats/trx.py", line 41, in read_results
    classname = test_classes[test_name]
KeyError: 'Microsoft.AspNetCore.Mvc.ApplicationModels.AttributeRouteModelTests.ReplaceTokens_InvalidFormat(template: "text]", values: [], reason: "Token delimiters (\'[\', \']\') are imbalanced.")'
+ date -u +%FT%TZ
2019-04-30T08:55:31Z
+ /usr/local/bin/python2 /Users/dotnet-bot/dotnetbuild/work/4bc49cfb-4e54-4e80-b956-4085f63855a8/Payload/xunit-reporter.py
2019-04-30 01:55:32,174: ERROR: xunit-reporter(69): main: Unable to report xunit results: no test results xml file found.
```

Maybe the uploading of the results went awry.

**Comment by alexperovich** [comment]
Ugh, thats just a bug with the trx parsing. Trx is sooo complicated for a simple type of data. I'll fix it.

**Comment by HaoK** [comment]
Cool, that's good news, well hopefully its just that bug on the parsing, and all our test data is already there

**Comment by alexperovich** [comment]
PR with fix: https://github.com/dotnet/arcade/pull/2648

**Comment by JunTaoLuo** [comment]
@alexperovich can you let me know which version of the sdk will have this fix when a build of it is available?

**Comment by alexperovich** [comment]
The build is currently waiting behind another build. I will let you know when a version containing the fix is available.

**Comment by alexperovich** [comment]
Version 2.0.0-beta.19230.6 has been published with the fix.

**Comment by JunTaoLuo** [comment]
Still missing some test results as Helix is only reporting from one of the .trx files. Will wait for another SDK update.

**Comment by JunTaoLuo** [comment]
Woot! I think all the test results are being reported now, thanks @alexperovich for the many fixes. I think after a final round of tests, this will be ready to get checked in!

**Review by natemcmaster — APPROVED** [review]

File: src/Servers/Kestrel/test/FunctionalTests/Http2/ShutdownTests.cs [hunk 10]
```diff
@@ -23,6 +23,7 @@ namespace Microsoft.AspNetCore.Server.Kestrel.FunctionalTests.Http2
 {
     [OSSkipCondition(OperatingSystems.MacOSX, SkipReason = "Missing SslStream ALPN support: https://github.com/dotnet/corefx/issues/30492")]
     [MinimumOSVersion(OperatingSystems.Windows, WindowsVersions.Win10)]
+    [SkipOnHelix(Queues = "Debian.8.Amd64.Open")] // Debian 8 uses OpenSSL 1.0.1 which does not support HTTP/2
```
Comment (JunTaoLuo) [inline src/Servers/Kestrel/test/FunctionalTests/Http2/ShutdownTests.cs]: Note to self: double check the versino of OpenSSL on Debian 8 agents and see if there's a way to manually update the version to one that supports HTTP/2

**Pull request merged**